server/vulnerabilities/msrc/io/msrc.go

```diff
@@ -0,0 +1,83 @@
+package io
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/pkg/download"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+)
+
+const (
+	// Pre 2020 there are some weirdness around the way the 'Supersedes' field is defined for Vulnerabilities, sometimes
+	// it does not reference a KBID.
+	MSRCMinYear = 2020
+	mSRCBaseURL = `https://api.msrc.microsoft.com`
+)
+
+// MSRCAPI allows users to interact with MSRC resources
+type MSRCAPI interface {
+	GetFeed(time.Month, int) (string, error)
+}
+
+type MSRCClient struct {
+	client  *http.Client
+	workDir string
+	baseURL *string
+}
+
+// NewMSRCClient returns a new MSRCClient that will store all downloaded files in 'workDir' and will
+// use 'baseURL' for doing http requests. If no 'baseURL' is provided then 'MSRCBaseURL' will be used.
+func NewMSRCClient(
+	client *http.Client,
+	workDir string,
+	baseURL *string,
```

Comment:
why is `baseURL` a pointer?

Reply:
To make this testable: https://github.com/fleetdm/fleet/pull/7491/files/c8276dd3ae701158ceb32af626f623ce68e68f5e#diff-280fa637863fb407769aea11e38af998b3839dc72c474a725150a352986fadf3R44

Reply:
Oh my bad, the question was `Why is this a pointer?` :rofl: - No reason really, I'll change it to a string.